src/main/java/org/apache/commons/io/input/BrokenInputStream.java

```diff
@@ -21,26 +21,43 @@
 import java.util.function.Supplier;
 
 /**
- * Always throws an {@link IOException} from all the {@link InputStream} methods where the exception is declared.
+ * Always throws an {@link IOException} or a {@link RuntimeException} from all {@link InputStream} methods where {@link IOException} is declared.
  * <p>
- * This class is mostly useful for testing error handling.
+ * This class is mostly useful for testing error handling in code that uses an {@link InputStream}.
  * </p>
  *
  * @since 2.0
  */
 public class BrokenInputStream extends InputStream {
 
     /**
-     * The singleton instance.
+     * A singleton instance using a default IOException.
      *
      * @since 2.12.0
      */
     public static final BrokenInputStream INSTANCE = new BrokenInputStream();
 
     /**
-     * A supplier for the exception that is thrown by all methods of this class.
+     * Constructs a new stream that always throws a {@link RuntimeException}.
```

Comment:
For testing error handling, it's useful to have a fake like this that does something a real stream might do, e.g. throwing an IOException on read.

It's not so useful to have a mock that's doing something no real stream should be doing like throwing a NullPointerException on read.

Reply:
The case that motivated me to raise this was testing code that is passed an arbitrary implementation of `Reader`.  

`Reader`s that throw `NullPointerException` on read might go against the intention of checked exceptions, but can exist, both intentionally and inadvertently.  Some methods on `Reader` also throw `IllegalArgumentException` and `UnsupportedOperationException`, though I concede that those are more the result of client errors than the operation of the `Reader` itself.

Reply:
Actually, I need to correct myself.  `java.io.Reader` itself never throws `UnsupportedOperationException`.  The only runtime exceptions its methods throw are `IllegalArgumentException` and `NullPointerException`, during validation of arguments.